libs/designsystem/menu/src/menu.component.ts

```diff
@@ -82,26 +86,123 @@ export class MenuComponent implements AfterViewInit, OnDestroy {
     | undefined;
 
   @ViewChild(FloatingDirective)
-  private floatingDirective: FloatingDirective;
+  _floatingMenu: FloatingDirective;
 
-  public FloatingOffset: typeof FloatingOffset = FloatingOffset;
+  @ContentChildren(ItemComponent, { read: ElementRef }) public kirbyItems: QueryList<
+    ElementRef<HTMLElement>
+  >;
 
+  public floatingMenuIsShown: boolean = false;
+  public FloatingOffset: typeof FloatingOffset = FloatingOffset;
   private scrollListenerDisposeFn: EventListenerDisposeFn;
+  private focusedIndex = -1;
+
+  @HostListener('keydown', ['$event'])
+  _onKeydown(event: KeyboardEvent) {
+    if (this.floatingMenuIsShown) {
+      this.handleOnKeyForOpenedMenu(event);
+    } else {
+      this.handleOnKeyForClosedMenu(event);
+    }
+  }
+
+  private preventFurtherPropagation(event: KeyboardEvent) {
+    event.stopImmediatePropagation();
+    event.preventDefault();
+  }
+
+  private handleOnKeyForClosedMenu(event: KeyboardEvent) {
+    const key = event.key;
+    switch (key) {
+      case ' ':
+      case 'Enter':
+      case 'ArrowDown':
+      case 'Down':
+        this.focusedIndex = 0;
+        this._floatingMenu.show();
+        this.focusItem();
+        this.preventFurtherPropagation(event);
+        break;
+      case 'Up':
+      case 'ArrowUp':
+        this.focusedIndex = this.kirbyItems.length - 1;
+        this._floatingMenu.show();
+        this.focusItem();
+        this.preventFurtherPropagation(event);
+        break;
+    }
+  }
+
+  private handleOnKeyForOpenedMenu(event: KeyboardEvent) {
+    const key = event.key;
+    switch (key) {
+      case 'ArrowDown':
+        if (this.focusedIndex < this.kirbyItems.length - 1) {
+          this.focusedIndex++;
+          this.focusItem();
+          this.preventFurtherPropagation(event);
+        }
+        break;
+      case 'ArrowUp':
+        if (this.focusedIndex > 0) {
+          this.focusedIndex--;
+          this.focusItem();
+          this.preventFurtherPropagation(event);
+        }
+        break;
+      case 'Escape':
+        if (this.closeOnEscapeKey) {
+          this.closeMenu();
+        }
+        this.preventFurtherPropagation(event);
+        break;
+      case 'Tab':
+        this.closeMenu();
+        break;
+    }
+  }
+
+  private focusTriggerButton() {
+    this.getTriggerButton().nativeElement.focus();
+  }
+
+  private closeMenu() {
+    this.resetFocus();
+    this._floatingMenu.hide();
+    this.focusTriggerButton();
+  }
+
+  resetFocus() {
+    this.focusedIndex = -1;
+  }
+
+  focusItem() {
+    const itemToBeFocused = this.kirbyItems.get(this.focusedIndex);
+    itemToBeFocused.nativeElement.focus();
```

Comment:
Focus the kirby-item which will then proxy down the focus to the wrapping button element within the ion-item.